woocommerce/woocommerce-android
# [STP] Adds a screen with QR that encodes sharable payment url

Closes: #8988


### Description

The PR adds a dialog that shows a QR code that encodes the payment URL. Before showing the QR we have to move order to the "pending" status otherwise it's not collectable

### Testing instructions

The may flow:
* On debug built type start collecting payment (both simple and order)
* Notice a new "scan to pay" button
* Click on it
* Notice the QR code and scan it
* Pay it online
* Notice that the order has been paid

Others:
* Simulate order update error (e.g. by disabling network on the device( - notice snack back with text shown and QR is not shown
* Also, validate that there is no regression with the "share payment URL" payment option

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/4923871/929a1610-2442-4307-9221-a11156ee04a4
